components/camera/videosource/helpers.go

```diff
@@ -22,6 +23,11 @@ import (
 // detected by http.DetectContentType.
 func getMIMETypeFromData(ctx context.Context, data []byte) (string, error) {
 	detectedMimeType := http.DetectContentType(data)
+	if !strings.Contains(detectedMimeType, "image") {
+		errMsg := errors.Errorf("%s", string(data))
```

Comment:
Hmm I take this back now. Given this is a bug, we should just say this is not an image MIME we will be able to decode. Taking all the data is a risk for logging garbage data in a terminal or revealing sensitive data.

Reply:
@tessavitabile suggested we check that the MIME is text/plain before attempting to decode it. Would that work? It'll ensure that we 
1. aren't displaying garbage in the terminal 
2. aren't revealing sensitive data since that info shouldn't be sent over plain text.

Reply:
I don't think looking for plain matters. It can still be a lot of text. We can just say we don't support non image MIMEs.

Reply:
that sounds fine